### ISSUE TITLE
`load_generated_functions` always puts generated functions in `Main` module

### ISSUE THREAD

**Issue by georgematheos** [probcomp/Gen.jl#209]
Sometimes we want to work with a Gen model in a module other than Main.  However, currently `Gen.load_generated_functions` is hardcoded to load the functions in the `Main` module.

This means if I do something along the lines of
```julia
module M1
using Gen
@gen (static) function ()....
...
end
Gen.load_generated_functions()
...
Gen.update(trace, ...)
end
```
we sometimes get an `UndefVarError: NoChange not defined`.  This originates from pretty deep in the macroexpansion for static trace update, and is a mess to understand (debugging these macros is painful!). It looks to me like the basic issue is that when we call `load_generated_functions`, we try to macroexpand some functions in the `Main` module, but these functions try to call `Gen.NoChange()`.  However, Gen is only imported into `M1`, not into `Main`.

***
Proposed solution:
Change `load_generated_functions` to `@load_generated_functions`.  Then instead of macroexpanding everything into `Main`, we can expand into `__module__` (ie. whatever module `@load_generated_functions` was called from).  This should be an easy change but will require updating the documentation.

**Comment by ztangent** [comment]
I'd appreciate this changing / extending this behavior as well, especially now that I have a fair amount of library code written as static generative functions!

Another possible solution would be to preserve the existing behavior, but the add another method for `load_generated_functions` which takes a `module` argument:

```julia
function load_generated_functions(__module__::Module)
    for function_defn in generated_functions
        __module__.eval(function_defn)
    end
end
```

And you'd use it as either:
```julia
module M1
using Gen
load_generated_functions(M1)
```
or
```julia
module M1
using Gen
load_generated_functions(@__MODULE__)
```
if you want to avoid retyping the module name.

@marcoct any thoughts on the above solution? And also, what do you think the default behavior should be? It seems more intuitive to me that by default we should eval in the calling module, as opposed to eval-ing in Main. I don't think changing the default behavior would actually be a breaking change in existing code either, since most of the time  `load_generated_functions` is called in a script, so it would still eval in Main.

**Comment by ztangent** [comment]
This is addressed by #234 :)
